test/lib/yggdrasil/command/parser_builder_test.exs

```diff
@@ -0,0 +1,58 @@
+defmodule TestCommand do
+  defstruct foo: nil, bar: nil, baz: nil
+end
+
+defmodule TestParser do
+  use Yggdrasil.Command.ParserBuilder
+
+  parse "test :foo", TestCommand
+  parse "test :foo :bar", TestCommand
+  parse "test :foo :bar :baz", TestCommand
+  parse "test :foo if :bar this :baz matters", TestCommand
```

Comment:
@copenhas I assume this was your intent, and had no idea that was possible till I studied the ParseRule a bit closer and started testing it.

All I have to say brilliant, works well. I had wondered if we were going to support this sort of thing, cause the move commands were simple and it was the only use of it I saw.